public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/DimensionFields.tsx

```diff
@@ -33,12 +82,22 @@ const DimensionFields: React.FC<DimensionFieldsProps> = ({ query, dimensionOptio
     onQueryChange(setDimensionFilterValue(query, filterIndex, fieldName, value));
   };
 
-  const onFilterInputChange = (index: number, ev: React.FormEvent) => {
-    if (ev.target instanceof HTMLInputElement) {
-      onFieldChange(index, 'filter', ev.target.value);
+  const onFilterInputChange = (index: number, v: SelectableValue<string>) => {
+    if (v) {
+      onFieldChange(index, 'filter', [v.value!]);
```

Comment:
We should avoid using the non-null assertion operator. If the value is nullable, we should check that this is not the case.

Reply:
Corrected, thanks 😊